compiler/grainformat/walktree.re

```diff
@@ -119,10 +119,22 @@ let walktree =
 
   all_locations := comment_locations;
 
-  let iter_location = (self, location) =>
-    if (!List.mem(Code(location), all_locations^)) {
-      all_locations := List.append(all_locations^, [Code(location)]);
+  // use a hash table to de-deupe adding locations to the list
```

Comment:
Any reason not to form a list from the hash when you need it?

Reply:
Oh, no reason!  I didn't realise you could get an ordered list out!